packages/react-server/src/ReactFlightServer.js

```diff
@@ -479,6 +560,31 @@ export function resolveModelToJSON(
         return serializeByValueID(errorId);
       }
     }
+    switch ((value: any).$$typeof) {
+      case REACT_PROVIDER_TYPE: {
+        const providerKey = ((value: any): ReactProviderType<any>)._context
+          ._globalName;
+        const writtenProviders = request.writtenProviders;
+        let providerId = writtenProviders.get(key);
+        if (providerId === undefined) {
+          request.pendingChunks++;
+          providerId = request.nextChunkId++;
+          writtenProviders.set(providerKey, providerId);
+          emitProviderChunk(request, providerId, providerKey);
+        }
+        return serializeByValueID(providerId);
+      }
+      case REACT_SERVER_CONTEXT_TYPE: {
+        if (key === '__pop') {
```

Comment:
I'm not sure when in the refactors this happened but this is not quite safe now.

Because if you passed something like `<ClientComponent __pop={context} />` it would pop.

It might be better to let the value be a magical value like a specific object reference that the user can't provide.

You don't need the context in the value anyway since it's just a DEV only safety thing you can remove that part from `popProvider`.

Reply:
That way you don't even have to check the key.

You can just do `if (value === POP)`.